addons/mail/static/src/core/common/res_partner_model.js

```diff
@@ -117,6 +118,25 @@ export class ResPartner extends Record {
         );
     }
 
+    displayDatetime() {
+        const currentUserTz = this.store.self_partner?.tz;
+        if (
+            !(this.realtimeDatetime && this.stopRealtimeTzDiff) &&
+            this.tz &&
+            currentUserTz &&
+            this.tz !== currentUserTz
+        ) {
+            this.stopRealtimeTzDiff = showRealtimeTzDiff(
+                currentUserTz,
+                this.tz,
+                ({ otherUserTime, otherUserDate }) => {
+                    this.realtimeDatetime = `${otherUserDate ?? ""} ${otherUserTime} local time`;
+                }
+            );
+        }
+        return this.realtimeDatetime;
+    }
```

Comment:
I'm confused by all of these functions and how they are spread in the code and how they all depend on each other.

What would be the easiet, probably: define 1 getter in partner, which returns the correct value for "now", regardless of the start/stop real time stuff.

Define a simple hook in component, which has a local state with the real time value, and a timeout to update it whenever necessary (and clear on unmount). Use that state in the template.

Avoid `useEffect` at all costs, as this hook overused and is not fit for state management.